### DISCUSSION TITLE
How will map updates be handled when currently on metered network

### DISCUSSION THREAD

**Discussion by Lineflyer** [cgeo/cgeo#12028] [Q&A] [answered]
Can somehow help me understand:
- I started an individual route while abroad and without WiFi at that moment
- c:geo asked me, whether I want to download missing routing information
- I confirmed but failed to activate "metered network" downloading, so nothing happend right away

What will happen now?
- Is this queued somewhere for download once an non-metered network is available?`
- Will it survive closure of c:geo and/or device restart?
- How could I cancel this decision and allow download over metered networks?

From my experience:
I could not see any remainders/queue of that action. After connecting to a WiFi nothing (visible) happend, so I force closed c:geo and invoked routing again to force c:geo to show that dialog again.
I just would like to know (also to answer support mails) how this is designed to work ;)

**Comment by moving-bits** [comment, marked as answer]
c:geo initiates the download using the system's download manager. If you're on a metered network, but haven't ticked the checkbox, c:geo will still fire up the download manager, and it depends on Android on when to restart it after wifi is available again.

To see what's currently scheduled by c:geo, you can open settings - system - view downloads. (It's just a rough display mainly intended for debugging.)

**Comment by Lineflyer** [comment]
Will c:geo handle it also after some hours (if its no longer running)?

Will take a look at that debugging though.

**Reply by moving-bits** [reply]
To my understanding the system should remember the request (and c:geo does internally by storing the additional request data in the `extension` table).
